libs/cpp-ledger/ledger/device_hid.cpp

```diff
@@ -0,0 +1,173 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "cpp-ledger/ledger/device_hid.hpp"
+
+#include "cpp-ledger/ledger/const.hpp"
+#include "cpp-ledger/ledger/utils.hpp"
+
+namespace ledger {
+
+  DeviceHid::DeviceHid() {
+    info.path = nullptr;
+    info.serial_number = nullptr;
+    info.manufacturer_string = nullptr;
+    info.product_string = nullptr;
+    info.next = nullptr;
+  }
+
+  DeviceHid::~DeviceHid() {
+    hid_close(device);
+
+    if (info.path != nullptr) {
+      delete[] info.path;
```

Comment:
hidapi uses `free`, not sure `delete[]` is valid/compatible

Reply:
hidapi uses free because it uses `calloc` for memory allocating.
Fields of `info` are created in `SetInfo` by calling `copyStr()` that uses `new[]`. In this case we must to use `delete[]`

Reply:
Btw, the approach could be totally changed